book/src/struct.md

```diff
@@ -319,6 +319,51 @@ Person_name <- function(self) {
 }
 ```
 
+It's important to mention that savvy only wraps the `EXTPTRSXP` in a closure
+environment when the type is used directly as the returning type of the function.
+If the user wants to return `Person` inside a `List`, for example, the external
+pointer will be directly exposed to the user and it will be the user's responsibility
+to deal with it.
+
+```rust
+#[savvy]
+struct Person {}
+
+// This case savvy handles nicely.
+/// @export
+#[savvy]
+impl Person {
+    fn new() -> savvy::Result<Person> {
+        Ok(Person {})
+    }
+}
+
+// In this case, the user is handled an external pointer.
+/// @export
+#[savvy]
+fn create_list() -> savvy::Result<Sexp> {
+    let mut list = OwnedListSexp::new(1, false)?;
+    let person = Person {};
+    list.set_value(0, Sexp::try_from(person)?)?;
+    list.into()
+}
+```
+
+in R:
+
+```r
+> person = Person$new()
+> print(person)
+<environment: 0x0000027cf9d46a20>
+attr(,"class")
+[1] "Person"
+
+> l = create_list()
+> print(l)
+[[1]]
+<pointer: 0x0000000000000001>
+```
+
```

Comment:
In my opinion, the case like this should be handled by creating another struct that holds the structs, for example,

```rust
struct Persons {
    person_list: Vec<Person>
}
```

or some wrapper on R's side. I'll add some more words when I come up with.

Reply:
Creating another struct that holds the struct is fine although that can involve move semantics to retrieve the inner struct from inside the other from R. For example in my case I have a struct like this:
```
#[savvy]
struct HDecodedAudio {
    harray: HArray,
    sr: u32,
}
```

where `HArray` is an `Arc`. Removing the `HArray` from inside `HDecodedAudio` in R would need to clone the `HArray` (which I would like to avoid as much as possible in my case, since I use COW semantics) or use move semantics, invalidating the original `HDecodedAudio`.

In my case, I chose to clone the value, but also to provide a finalizer (`invalidate`) for the `HDecodedAudio`, so the reference count could go back to 1 if the user wants to, removing the shared reference after cloning.

```
#[savvy]
impl HDecodedAudio {
    fn harray(&self) -> savvy::Result<HArray> {
        self.harray.clone()
    }
    pub fn invalidate(self) -> savvy::Result<()> {
        Ok(())
    }
}
```

With List I wouldn't have to create this `invalidate` function and no move semantics would be needed.